### PR TITLE
add getDataByLanguage into index.d.ts

### LINKED ISSUE #1273: Is getDataByLanguage a public api?
Changelog mentions it under (11.6.0) 
https://github.com/i18next/i18next/blob/master/CHANGELOG.md#1160
but I don't see it in documentation or typescript definition file.

### PR DESCRIPTION
**Pull request by msftjayongg**
update index.d.ts with the public getDataByLanguage API interface.  Basically this allows typescript to call the API that already exists. No functionality change.

Closes #1273

### CODE REVIEW

**Comment by msftjayongg** [comment]
@rosskevin I was told to tag you.

I am also unsure how to actually run the typescript tests - I guess they are there just for tslint purposes? (I'd feel better if I could run it :) )

File: test/typescript/init.test.ts [hunk 1]
```diff
@@ -205,6 +205,12 @@ const v: string = i18next.t('my.key');
 const de = i18next.getFixedT('de');
 const z: string = de('myKey');
 
+// verify we can get the German data
+const data = i18next.getDataByLanguage('de');
+if (data.translation.myKey !== de('myKey')) {
```
Comment (rosskevin) [inline test/typescript/init.test.ts]: This is type usage test only, so we only need the line 209, please remove the `if` block to keep these tests as brief as possible.

**Comment by msftjayongg** [comment]
Updated - I left a line such that we can check that the .translation field exists.

File: test/typescript/init.test.ts (hunk 1 above)
Comment (msftjayongg) [reply test/typescript/init.test.ts]: updated with a change - please look :)

**Comment by rosskevin** [comment]
Thanks @msftjayongg 

@jamuhl this can be a patch release.

**Comment by jamuhl** [comment]
published in i18next@17.0.16

**Pull request merged**